sgx-tools/gen-quote.go

```diff
@@ -21,13 +22,21 @@ For example, generate the quote file according to the given local report file:
 
 	# sgx-tools gen-quote --report foo.rep --spid ${SPID}`,
 	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "serviceType",
+			Usage: "specify the aesmd service type such as DCAP or EPID",
+		},
```

Comment:
Because sgx-tools always runs on a genuine machine so there is knowledge to decide which one is really used. For example, we hope to use sgx-tools to generate a quote and don't care the quote is based on ecdsa or epid. This is the smart way to implement.

Reply:
> Because sgx-tools always runs on a genuine machine so there is knowledge to decide which one is really used. For example, we hope to use sgx-tools to generate a quote and don't care the quote is based on ecdsa or epid. This is the smart way to implement.

We must first reconstruct the targetinfo and quote of epid to realize this function.

Reply:
> > Because sgx-tools always runs on a genuine machine so there is knowledge to decide which one is really used. For example, we hope to use sgx-tools to generate a quote and don't care the quote is based on ecdsa or epid. This is the smart way to implement.
> 
> We must first reconstruct the targetinfo and quote of epid to realize this function.

I think the sub pr to rune/libenclave/intelsgx package(sgx-tools/dcap_linux.go、sgx-tools/proto/aesm-service.proto、sgx-tools/proto/aesm-service.pb.go) should be reviewed  and merged in short term. For the sub pr to sgx-tools(sgx-tools/gen-qe-target-info.go、sgx-tools/gen-quote.go、sgx-tools/utils_linux.go) will be not merged in short term.